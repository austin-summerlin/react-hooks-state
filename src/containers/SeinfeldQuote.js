/* eslint-disable max-len */
import React, { useState } from 'react';
import Load from '../components/load/Load';
import Quote from '../components/quote/Quote';
import { getSeinfeldQuote } from '../services/seinfeldApi';
import style from '../containers/SeinfeldQuote.css';

export default function SeinfeldQuote() {
  const [quote, setQuote] = useState({});
  const [loading, setLoading] = useState(false);

  const handleClick = async () => {
    const quote = await getSeinfeldQuote();
    setQuote(quote);
    setLoading(false);
  };
  return (
    <div className={style.root}>
      <Load onClick={handleClick} />
      {loading ? <h1 className={style.loading}>Loading...</h1> : <Quote className={style.quote} {...quote} />}
    </div>
  );
}

