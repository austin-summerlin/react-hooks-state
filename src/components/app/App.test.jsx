import React from 'react';
import { render, cleanup } from '@testing-library/react';
import SeinfeldQuote from '../../containers/SeinfeldQuote';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<SeinfeldQuote />);
    expect(asFragment()).toMatchSnapshot();
  });
});
