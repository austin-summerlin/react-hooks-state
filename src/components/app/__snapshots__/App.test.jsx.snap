// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div>
      <h2
        class="title"
      >
        Seinfeld
      </h2>
      <button
        class="loadButton"
      >
        Get Quote
      </button>
    </div>
    <div
      class="quote"
    >
      <figure>
        <h3 />
        <p />
      </figure>
    </div>
  </div>
</DocumentFragment>
`;
